src/track/trackrecord.h

```diff
@@ -16,17 +16,17 @@ class TrackDAO;
 namespace mixxx {
 
 /// Effect of updating a property with a new value.
-enum class [[nodiscard]] UpdateResult{
-        /// The value has been updated and changed.
-        Updated,
+enum class UpdateResult {
+    /// The value has been updated and changed.
+    Updated,
 
```

Comment:
This tells me that the enum was deliberately made for error handling. Specifically `Rejected` should be checked because it almost always would require handling from the caller. This removes that feature, making the error checking optional (which I hope we can agree is very bad practice). 

I need more explanation why its appropriate to weaken the guarantees this enum was supposed to provide.

Reply:
Concretely: What is your rationale for removing the checks you removed in this PR?

Reply:
We have discussed this earlier and conclude that `[[nodiscard]]` is usefull if discarding is an error. In this case it is only an enum. Discarding can't be an error. It might be an error depending on the returning function. In our case it is `updateGlobalKeyNormalizeText()` wich works well without error checking. So the change allows us to remove `static_cast<void>()` on these calls. See at [dlgtrackinfomulti.cpp](https://github.com/mixxxdj/mixxx/pull/14022/files#diff-39b5603d1dbe5d82fe8852e827f9291cbb8d3d6cbe297992922ec54620e971cc)

The counter example are places like this where `[[nodiscard]]` heps to avois programming errors: https://github.com/mixxxdj/mixxx/blob/a4e61f9d245079d0f00aabf9e2876c1e9285679f/src/util/compatibility/qmutex.h#L32

Reply:
> In this case it is only an enum. Discarding can't be an error.

Is it though? It seems that `Rejected` and `Unchanged` did need some special handling in some cases (which you have removed). I'm asking specifically why its fine to remove the handling for those cases here.

Reply:
I think so. I have left an inline comment: 
https://github.com/mixxxdj/mixxx/pull/14022#discussion_r1888507649